flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/AbstractPostgresCompatibleDialect.java

```diff
@@ -56,6 +56,7 @@ public Optional<String> getUpsertStatement(
                         .collect(Collectors.joining(", "));
         String updateClause =
                 Arrays.stream(fieldNames)
+                        .filter(f -> !Arrays.asList(uniqueKeyFields).contains(f))
```

Comment:
Would you mind elaborate on why adding this filter?

If adding this filter makes sense, I left 2 comments about code logic:
- We can initialize the `Arrays.asList(uniqueKeyFields)` only once. IIUC, this array will be converted to List multiple times when `fieldNames` length > 0.
- We call the contains method, it's better to use Set instead of List. right?

Reply:
Thanks @1996fanrui , I think that was a little bit inaccurate, update clause must exclude unique index . If not will case execptions like 'modification of distribution columns in OnConflictUpdate is not supported Call getNextException to see other errors in the batch'.